### DISCUSSION TITLE
Meeting note #5, 2022-11-30

### DISCUSSION THREAD

**Discussion by jongfeel** [fkdl0048/Mentoring-Project#24] [General]
### 팀 커뮤니케이션

같이 개발 하는 분과 개발 분담 협의
기획, 아트 쪽 분과도 잘 얘기해서 진행

### 개인 프로젝트에 대한 의견

개인 프로젝트는 의지를 많이 넣어야 하므로
반드시 작은 범위를 잡고 진행해야 함

개발한 결과를 사용자가 쓰는 물건으로 만든다면
상당히 좋은 소프트웨어를 만드는 경험을 할 수 있음

1순위는 사용자(고객)가 원하는 걸 만들어야 하는 것이고,
2순위가 그 원하는 걸 퀄리티 있게 만든다가 중요

Unity editor를 사용해서 특정 기능을 구현하는 건 개인 프로젝트로 하기에는 상당히 좋은 범위
조금 더 욕심을 내서 unity package manager에 추가할 수 있는 형태의 패키지로 만들어서 배포해 보는 경험도 좋음
(거의 google play store에 등록하는 것과 비슷한 느낌)

### 기타

책 전달 장소 협의

**Comment by fkdl0048** [comment]
## 12/02 초기 기획

기존 스토어 출시를 위한 어플 개발 보다 2달간 단기 개발에 적합한 유니티에서 사용가능한 커스텀 에디터를 패키지로 배포하고자 함.  

1. 기존 어플에 대한 불확실 성 및 배포나 어플 개발에 성공 경험이 X 따라서 성취도를 올리고자 함
2. UIToolKit을 게임잼 프로젝트에 적용해본 결과 재사용성이 매우 뛰어나지만 아직은 개발단계, 그럼에도 유니티에서 밀고있는 3세대 UI이기 때문에 유용해보임
3. Editor기능을 좀 더 이해하고 유니티에 대한 시야 자체도 높아질 것으로 보임

### Player Input Saver (가제)

**원하는 인풋을 녹화하거나 미리 입력하여 조절 수정이 가능한 에디터**

기본적으로 기획과 개발단계에서 사용될 것으로 예상된다.  

> 예상하는 기능들

1. 플레이에 앞서 원하는 키(점프, 움직임 등)다양한 키값을 입력하여 실행하는 경우 자동으로 인풋값을 넣는다.(실행 시 자동으로 실행)  
2. 플레이하기 전 값을 녹화하여 끝난 후 해당 값을 확인 또는 수정 후 재실행이 가능해진다.  


> 예상하는 효과  

기획이나 개발단계에서 특정상황에 발생하는 버그를 영상 녹화가 아닌 인풋값 녹화를 통해 디버깅이 가능해짐  

좀 더 유동적으로 1프레임 씩 재생하며 애니메이션 체킹이 가능해짐

#### 개발해야 하는 기능  

1. 입력값에 대한 파싱 처리
2. 수정에 대한 에러 처리

**Reply by fkdl0048** [reply]
간단하게 기획만 해놓고 일단 진행해보려고 합니다..!
